metaflow/datastore/task_datastore.py

```diff
@@ -435,6 +435,37 @@ def save_metadata(self, contents, allow_overwrite=True, add_attempt=True):
             add_attempt,
         )
 
+    @require_mode("w")
+    def _dangerous_save_metadata_post_done(
+        self, contents, allow_overwrite=True, add_attempt=True
+    ):
+        """
+        Method identical to save_metadata BUT BYPASSES THE CHECK ON DONE
+
+        @warning This method should not be used unless you know what you are doing. This
+        will write metadata to a datastore that has been marked as done.
+
+        Currently used to re-patriate metadata in the case of a remote run and no
+        metadata service
```

Comment:
Maybe there is a way to phrase this in a way that is easier to understand for someone new to the codebase, see my take below

```suggestion
        Currently only used in the case when the task is executed remotely but there is no (remote) metadata service configured, and we use datastore to share metadata between the task and the Metaflow local scheduler
```